classes/mail/Mailable.php

```diff
@@ -359,6 +359,9 @@ protected function buildSubject($message): self
             ->toArray();
 
         $subject = app('mailer')->compileParams($this->subject, $withoutTagViewData);
+        // decode HTML entities for the subject
+        $subject = htmlspecialchars_decode($subject, ENT_QUOTES | ENT_HTML5);
+
         if (empty($subject)) {
             trigger_error(
                 'You are sending ' . static::getName() ?? static::class . ' email with empty subject',
```

Comment:
@Hafsa-Naeem as you're updating this part of code, could please fix this issue here?

The second part of the `??` will never be executed.

Reply:
@Hafsa-Naeem, I see you've changed this line, but I don't think the change addresses @jonasraoni's comment. I think he was pointing out that `Mailable::getName()` will always return a `string` (see [the function declaration](https://github.com/pkp/pkp-lib/blob/1d2e1eeedde89ef2a42d687e1a3cd69ad61db39c/classes/mail/Mailable.php#L134)), so the `?? static::class` will never be executed. You can remove `?? static::class` entirely.

Reply:
@jonasraoni and @Hafsa-Naeem, it looks like you two collided on this change:
- @jonasraoni made this change: https://github.com/pkp/pkp-lib/commit/49ad7a8a02ad34dbc06ad148224bd8ac5d42ccfd
- @Hafsa-Naeem made this change: https://github.com/pkp/pkp-lib/commit/70278465c9c0137a79bc4092ee064e67c64ffe2d

@jonasraoni, what was the intention of your change?

Reply:
Hi! I went ahead and applied the blind fix yesterday, but after watching the code...

I think we should use just the `static::class`. The `getName()` uses translated content, but logs are for developers, we don't need it, it just makes things more difficult to understand (e.g. user might change the translation). If the locale is important for the error, it should be added properly.

But if we want to keep it this way... Then as the `getName()` returns a string, and possibly an empty one, it would be better to use `(static::getName() ?: static::class)`.

And a last comment: The `trigger_error()` isn't very helpful, I'd rather use an `Exception` (`error_log(new Exception('error'))`) to display the stack trace, until we enable the Laravel logger https://github.com/pkp/pkp-lib/discussions/7966.

Reply:
@asmecher @jonasraoni Thanks for the feedbacks. based on both your reviews, the solution might be:

- Retaining the initial check for null and set this->subject to an empty string if needed.
- Use static::class for logging since getName may return translated content which might not be useful in logs.

- Also replace trigger_error with error_log(new Exception(...)) for better debugging?

```
if (empty($subject)) {
    error_log(new Exception(
        'You are sending ' . static::class . ' email with empty subject'
    ));
}

```

Reply:
We're bikeshedding at this point :)

This code is only executed in an unusual circumstance, and we just need something that'll allow us to help debug when someone in the wild encounters it e.g. with custom code or an out-of-spec database. I'm perfectly happy with the approach that's currently in the PR, @Hafsa-Naeem.

However, we do have a merge conflict. I'll revert https://github.com/pkp/pkp-lib/commit/49ad7a8a02ad34dbc06ad148224bd8ac5d42ccfd and merge this PR as-is.